src/services/Provider.ts

```diff
@@ -192,20 +193,24 @@ export class Provider {
   /** Get URL details (if possible)
    * @param {string} url or did
    * @param {string} providerUri uri of the provider that will be used to check the file
+   * @param {boolean} withChecksum if true, will return checksum of files content
    * @param {AbortSignal} signal abort signal
+   * @param {any} headers headers key value pairs associated with the asset GET request
    * @return {Promise<FileInfo[]>} urlDetails
    */
   public async getFileInfo(
     file: UrlFile | Arweave,
     providerUri: string,
-    signal?: AbortSignal
+    withChecksum: boolean = false,
+    signal?: AbortSignal,
+    headers?: any
```

Comment:
can't we define a interface for this to avoid using `any` , with the desired HTTP get fields needed ?